config/management/operational/src/keys.rs

```diff
@@ -1,15 +1,159 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use diem_management::{config::ConfigPath, error::Error, secure_backend::ValidatorBackend};
-use serde::Serialize;
+use diem_client::AccountAddress;
+use diem_config::config::{Peer, PeerRole};
+use diem_crypto::{
+    ed25519::{Ed25519PrivateKey, Ed25519PublicKey},
+    x25519, Uniform, ValidCryptoMaterial, ValidCryptoMaterialStringExt,
+};
+use diem_global_constants::OWNER_ACCOUNT;
+use diem_management::{
+    config::{Config, ConfigPath},
+    error::Error,
+    secure_backend::ValidatorBackend,
+};
+use diem_types::{account_address::from_identity_public_key, PeerId};
+use rand::SeedableRng;
 use std::{
-    fs::File,
+    collections::{HashMap, HashSet},
+    fs::{read, File},
     io::Write,
     path::{Path, PathBuf},
+    str::FromStr,
 };
 use structopt::StructOpt;
 
+#[derive(Debug, StructOpt)]
+pub struct GenerateKey {
+    /// Location to store the key
+    #[structopt(long)]
+    key_file: PathBuf,
+    #[structopt(long)]
+    key_type: KeyType,
+    #[structopt(long)]
+    encoding: EncodingType,
+}
+
+#[derive(Debug, StructOpt)]
+pub enum KeyType {
+    Ed25519,
+    X25519,
+}
+
+impl FromStr for KeyType {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "Ed25519" | "ed25519" => Ok(KeyType::Ed25519),
+            "X25519" | "x25519" => Ok(KeyType::X25519),
+            _ => Err("Invalid key type"),
+        }
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub enum EncodingType {
+    BCS,
+    Hex,
+}
+
+impl FromStr for EncodingType {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "hex" | "Hex" => Ok(EncodingType::Hex),
+            "bcs" | "Bcs" | "BCS" => Ok(EncodingType::BCS),
```

Comment:
same... to lower saves the day

Reply:
could we add base64, because of reasons?

Reply:
Sure